aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,1105 @@
+        assert!(now > proposal.start_time_sec, error::invalid_argument(EPROPOSAL_NOT_STARTED));
+
+        let prop_stats = borrow_global_mut<ProposalVotingStatistics>(nft_dao);
+
+        // initialize the voting statistics of the proposal
+        if (!table::contains(&prop_stats.proposals, proposal_id)) {
+            let vstat = VotingStatistics {
+                total_yes: 0,
+                total_no: 0,
+                yes_votes: bucket_table::new(10),
+                no_votes: bucket_table::new(10),
+            };
+            table::add(&mut prop_stats.proposals, proposal_id, vstat);
+        };
+        let stats = table::borrow_mut(&mut prop_stats.proposals, proposal_id);
+
+        let voter_addr = signer::address_of(account);
+        let i = 0;
+        // loop through all NFTs used for voting and update the voting result
+        while (i < vector::length(&token_names)) {
+            let token_name = *vector::borrow(&token_names, i);
+            let property_version = *vector::borrow(&property_versions, i);
+            let token_id = token::create_token_id_raw(gtoken.creator, gtoken.collection, token_name, property_version);
+            // check if this token already voted
+            assert!(!bucket_table::contains(&stats.no_votes, &token_id), error::invalid_argument(ETOKEN_ALREADY_VOTED));
+            assert!(!bucket_table::contains(&stats.yes_votes, &token_id), error::invalid_argument(ETOKEN_ALREADY_VOTED));
+
+            // this account owns the token
+            assert!(token::balance_of(signer::address_of(account), token_id) == 1, error::permission_denied(ENOT_OWN_THE_VOTING_DAO_TOKEN));
+            if (vote) {
+                stats.total_yes = stats.total_yes + 1;
+                bucket_table::add(&mut stats.yes_votes, token_id, voter_addr);
+            } else {
+                stats.total_no = stats.total_no + 1;
+                bucket_table::add(&mut stats.no_votes, token_id, voter_addr);
+            };
+            i = i + 1;
+        };
+
+        nft_dao_events::emit_voting_event(
+            voter_addr,
+            nft_dao,
+            proposal_id,
+            vote,
+            token_names,
+            property_versions,
+        );
+    }
+
+    /// Admin can veto an active proposal
+    public entry fun admin_veto_proposal(admin: &signer, proposal_id: u64, nft_dao: address) acquires DAO, Proposals {
+        let dao = borrow_global_mut<DAO>(nft_dao);
+        assert!(dao.admin == signer::address_of(admin), error::permission_denied(EINVALID_ADMIN_ACCOUNT));
+        // assert the proposal is still active
+        let proposals = borrow_global_mut<Proposals>(nft_dao);
+        let proposal = table::borrow_mut(&mut proposals.proposals, proposal_id);
+        let now = timestamp::now_seconds();
+        assert!(now < proposal.start_time_sec + dao.voting_duration, error::invalid_argument(EPROPOSAL_ENDED));
```

Comment:
I don't think we need this check at all? Admin should be able to veto the proposals as long as it's still pending, regardless of whether the voting duration has ended or not. Same for resolve as well?

Reply:
removed all the time check for this flow. I think the admin should even be able to veto or resolve the proposal even before it starts